### ISSUE TITLE
Filter Name must be user-definable

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#540]
Some management profiles require certain values for the `Name` property of indication filters. 

However, the current implementation of the subscription manager sets the `Name` property to a certain format.

**Comment by andy-maier** [comment]
This is what DSP1054 1.2.2 says about the `Name` property of `CIM_IndicationFilter`:

---------------------------

**7.3.11.3.2 Property: Name**

The value of the Name property shall be the name of the indication filter; it shall be formatted as defined by the following ABNF rule:

```
    OrgID ":" RegisteredName ":" UniqueID
```

`OrgID` shall identify the business entity owning the referencing profile. `OrgID` shall include a copyrighted, trademarked, or otherwise unique name that is owned by that business entity or that is a registered ID assigned to that business entity by a recognized global authority. In addition, to ensure uniqueness, `OrgID` shall not contain a colon (:). For referencing profiles owned by DMTF, `OrgID` shall match "DMTF".

`RegisteredName` shall be the registered name of the referencing profile, as defined by the value of the RegisteredName property in the RegisteredProfile instance representing the implemented version of that profile.

`UniqueID` shall uniquely identify the represented indication filter within the referencing profile.

**DEPRECATED (begin)**

For compatibility with version 1.0 of this profile, referencing profiles owned by business entities other than DMTF may in addition define values for the Name property that are formatted as defined by the following ABNF rule:

```
    OrgID ":" UniqueID
```

Where:
* `OrgID` is defined above in this subclause.
* `UniqueID` shall uniquely identify the instance within the business entity owning the referencing profile.

Version 1.1 of this profile has deprecated this additional format.

**DEPRECATED (end)**

---------------------------

**Comment by andy-maier** [comment]
Karl to find out how SMI-S treats the `Name` property.

**Comment by andy-maier** [comment]
It seems we do not have any place in a filter where we can safely store a pywbem-specific identification:
- The `Name` property may be determined by profiles
- The other key properties (`SystemName` and the two class names) also have a defined semantics
- None of the other properties covered in the indications profile can be used

In theory we could use one of the properties defines on the class but not used in the indications profile, e.g. `ElementName`. The question is whether the server implementations support that.

**Comment by KSchopmeyer** [comment]
As of smis 1.7 SMI places not place any requirements on the name property above any beyond DSP1054. All of the specific information on the characteristics of the name property were removed.

However, in earlier versions of the spec, the name property is often defined explicitly allowing NO variation. Thus, for example, one particular filter in section 4.8.9 of the block books has the fixed string:

SNI:Array:SystemDeletion

as the Mandatory required string for the filter.

At the same time, I asked Mike Walker how we id static filters so that we can separate them from non static.  I am not certain, but it would appear that the only required diferences is that they cannot be deleted.  Need to check that.

Therefore, it would appear that we do have a real problem here.  We need to allow for the fixed name but any filter and corresponding subscription made this way must be out of the clients control completely because there is no way that the client can ever really insure that the indication was built by them. Of the fields that make up the key, (SystemCreationClassName, CreationClassName, SystemName, Name) there is nothing that in any way insures uniqueness or ability to really define who created the filter.  In fact,  thiis should probably have only been available as a static filter since creating a unique filter this way would be very difficult.

I propose that we consider the following:

1. Extend the API to allow creating a filter where the Name property is completely specified by the createfilter call.  Clarify the issues with this filter including the fact that it cannot be an owned filter.

2. To cover the fact that the general requirement is for org and registered name fields within the filter, add these to the API (ex. apis to set these before filters are created).

3. Keep our format for owned filters.

By the way, to get around all this confusion, some clients create two filters, one that they leave on the system with the fixed name and another one that has a unique name that they can control.  Apparently the create subscriptions for the unique one.  

NOTE: This does not answer the question about how we can determine if a filter is truly static.

**Comment by andy-maier** [comment]
In the proposal: I agree with 1. and 3. I'm not sure 2. is worth the effort.

**Comment by KSchopmeyer** [comment]
Karl agrees with Andy's proposal.

Andy suggests we bring this back to SNIA and DMTF to get this issue fixed in the spec.  Make new issue for the work to bring this back Karl

**Comment by andy-maier** [comment]
When I started to write the example for issue #462, it became clear to me that our add_*() functions always try to create the owned instances, even if we know in our instance variables that they must already exist (e.g. because we discovered them when adding the server).

We don't even nicely support finding out whether the desired instance exists, and leave that whole complexity entirely to the user.

We could significantly improve that by conditionally creating owned instances, only if they are not yet known in our instance variables.
